services/messaging/src/main/java/org/eclipse/hono/messaging/HonoMessagingMessageFilter.java

```diff
@@ -58,7 +57,7 @@ public static boolean verify(final ResourceIdentifier linkTarget, final Message
          if (msg.getContentType() == null) {
              LOG.trace("message [{}] has no content type", msg.getMessageId());
              return false;
-         } else if (msg.getBody() != null && !(msg.getBody() instanceof Data)) {
+        } else if (!MessageHelper.hasDataBody(msg, false)) {
```

Comment:
I think we still need the `null` check because an *empty notification* event may have an empty payload (as the name suggests) ...